### PR TITLE
Separate tests and coding standards

### PR DESCRIPTION
**Pull request by rhattersley**
See https://travis-ci.org/rhattersley/pyugrid/builds/109311275 for an example of a coding standards failure.

### CODE REVIEW

**Comment by ocefpaf** [comment]
This separation of context is awesome.  Thanks @rhattersley!

File: .travis.yml [hunk 1]
```diff
@@ -21,8 +25,12 @@ install:
     - mkdir files
 
 script:
-    - python setup.py test -a "--verbose"
-    - find . -type f -name "*.py" ! -name 'conf.py' ! -name 'ugrid_wx.py' | xargs flake8 --max-line-length=100
+    - if [[ $TEST_TARGET == 'default' ]]; then
+        python setup.py test -a "--verbose";
+      fi
+    - if [[ $TEST_TARGET == 'coding' ]]; then
+        find . -type f -name "*.py" ! -name 'conf.py' ! -name 'ugrid_wx.py' | xargs flake8 --max-line-length=100;
```
Comment (ocefpaf) [inline .travis.yml]: I think that #114 was fixed. So we can remove `ugrid_wx.py` from the ignore list and show what this PR can do here :wink:

Comment (rhattersley) [inline .travis.yml]: > we can remove ugrid_wx.py from the ignore list and show what this PR can do here :wink:

It works ... it failed! :grin:

**Comment by rhattersley** [comment]
> It works ... it failed! :grin:

And now, thanks to 43d892a7cf117eb8db96c929afd7caa59b0f9665, it [passes](https://travis-ci.org/pyugrid/pyugrid/builds/109318018).

Q.E.D. :smile:

**Comment by ocefpaf** [comment]
Awesome!  I like how Travis-CI reports this.  No need to scroll down the log to see the last failing test.

**Comment by ocefpaf** [comment]
I've been giving commit rights here. (Thanks @ChrisBarker-NOAA!)

Hope I am not abusing my powers by merging this one :wink:

**Comment by ChrisBarker-NOAA** [comment]
OK -- this looks pretty cool -- I"m interested to see how it works out.

Now I need to make sure I've got flake8 turned on in the editors on all my machines....

Thanks, @rhattersley -- really nice improvement

**Pull request merged**